### PR TITLE
luminous: mds: configurable timeout for client eviction

### PR DESCRIPTION
**Pull request by batrick**
http://tracker.ceph.com/issues/35976

### CODE REVIEW

File: src/mds/Locker.cc [hunk 1]
```diff
@@ -3519,7 +3519,8 @@ void Locker::remove_client_cap(CInode *in, client_t client)
  * Return true if any currently revoking caps exceed the
  * mds_session_timeout threshold.
  */
-bool Locker::any_late_revoking_caps(xlist<Capability*> const &revoking) const
+bool Locker::any_late_revoking_caps(xlist<Capability*> const &revoking,
+                                    double timeout) const
```
[the remark this answers is not in the document]
Comment (batrick) [reply src/mds/Locker.cc]: Great catch! Fixed.

**Comment by vshankar** [comment]
just needs a rebase...

**Comment by batrick** [comment]
done

**Comment by yuriw** [comment]
@batrick has conflicts

--- pr 24086 --- pulling https://github.com/batrick/ceph.git branch i35976
remote: Enumerating objects: 36, done.
remote: Counting objects: 100% (36/36), done.
remote: Total 42 (delta 36), reused 36 (delta 36), pack-reused 6
Unpacking objects: 100% (42/42), done.
From https://github.com/batrick/ceph
 * branch            i35976     -> FETCH_HEAD
Auto-merging src/mds/Server.cc
Auto-merging src/mds/MDSRank.h
Auto-merging src/mds/MDSRank.cc
Auto-merging src/mds/MDSDaemon.cc
Auto-merging src/mds/Beacon.cc
Auto-merging src/common/options.cc
CONFLICT (content): Merge conflict in src/common/options.cc
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r
AssertionError

**Comment by yuriw** [comment]
[wip-yuri-testing-2018-09-27-2343-luminous](https://trello.com/c/oRffqIRZ/420-wip-yuri-testing-2018-09-27-2343-luminous)

**Comment by batrick** [comment]
This one ahs a bug causing this failure

/ceph/teuthology-archive/yuriw-2018-09-28_12:39:34-fs-wip-yuri-testing-2018-09-27-2343-luminous-distro-basic-smithi/3079595/teuthology.log


will fix later today.

**Comment by yuriw** [comment]
[wip-yuri-testing-2018-10-01-2144-luminous](https://trello.com/c/P3SRyOZY/426-wip-yuri-testing-2018-10-01-2144-luminous)

**Comment by batrick** [comment]
````
2018-10-02T05:04:28.919 INFO:tasks.cephfs_test_runner:======================================================================
2018-10-02T05:04:28.919 INFO:tasks.cephfs_test_runner:ERROR: test_cap_revoke_nonresponder (tasks.cephfs.test_misc.TestMisc)
2018-10-02T05:04:28.919 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2018-10-02T05:04:28.919 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2018-10-02T05:04:28.919 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_wip-yuri-testing-2018-10-01-2144-luminous/qa/tasks/cephfs/test_misc.py", line 176, in test_cap_revoke_nonresponder
2018-10-02T05:04:28.920 INFO:tasks.cephfs_test_runner:    self.assertTrue(self.mount_a.is_blacklisted())
2018-10-02T05:04:28.920 INFO:tasks.cephfs_test_runner:AttributeError: 'FuseMount' object has no attribute 'is_blacklisted'
````
From: /ceph/teuthology-archive/yuriw-2018-10-02_01:20:54-fs-wip-yuri-testing-2018-10-01-2144-luminous-distro-basic-smithi/3092744/teuthology.log

will fix by removing that check

**Comment by yuriw** [comment]
[wip-yuri-testing-2018-10-02-2105-luminous](https://trello.com/c/KubwQ8Te/429-wip-yuri-testing-2018-10-02-2105-luminous)

**Comment by batrick** [comment]
`Reviewed-by: Patrick Donnelly <pdonnell@redhat.com>`

**Pull request merged**